src/components/pages/search.js

```diff
@@ -35,14 +37,26 @@ function Search({ initialState }) {
 
   const [filteredDocs, setFilteredDocs] = useState([...sortedDocs]);
 
-  const [selectedTags, setSelectedTags] = useState([]);
+  const queryTagsToLower = new Set(decodedQuery.split(/\s/)
+    .filter(tag => tag.length > 3)
+    .map(tag => tag.toLowerCase()));
+
+  const { allTags, tagMap } = docs.reduce((acc, doc) => {
+    doc.tags.forEach(tag => {
+      acc.allTags.add(tag);
+      acc.tagMap[tag.toLowerCase()] = tag;
+    });
 
-  const allTags = docs.reduce((acc, doc) => {
-    doc.tags.forEach(tag => acc.add(tag));
     return acc;
-  }, new Set());
+  }, { allTags: new Set(), tagMap: {} });
+
+  const initialSelectedTags = [...queryTagsToLower]
+    .map(tag => tagMap[tag])
+    .filter(tag => !!tag);
+
+  const [selectedTags, setSelectedTags] = useState(initialSelectedTags);
 
-  const tagOptions = Array.from(allTags).map(tag => ({ value: tag, key: tag }));
+  const tagOptions = [...allTags].map(tag => ({ value: tag, key: tag }));
```

Comment:
If you end up changing as I suggested, here you can replace `[...allTags]` with `allUniqueTags` as it's already an array.